### PR TITLE
Add an XLA utility function HasOverlappinWindow.

### PR DESCRIPTION
**Pull request by nouiz**
This is a split from PR: https://github.com/tensorflow/tensorflow/pull/32251

### CODE REVIEW

File: tensorflow/compiler/xla/window_util.h [hunk 1]
```diff
@@ -54,6 +54,7 @@ bool HasSymmetricPadding(const PaddingConfig& padding_config);
 bool HasBaseDilation(const Window& window);
 bool HasWindowDilation(const Window& window);
 bool HasDilation(const Window& window);
+bool HasOverlappinWindow(const Window& window);
```
Comment (cheshire) [inline tensorflow/compiler/xla/window_util.h]: Spelling, docstring missing.

**Review by cheshire — CHANGES_REQUESTED** [review]

**Comment by nouiz** [comment]
Fixed the comments.

File: tensorflow/compiler/xla/window_util.h [hunk 2]
```diff
@@ -55,6 +55,10 @@ bool HasBaseDilation(const Window& window);
 bool HasWindowDilation(const Window& window);
 bool HasDilation(const Window& window);
 
+// Return true if the window overlap (e.g., the window size is smaller
```
Comment (cheshire) [inline tensorflow/compiler/xla/window_util.h]: "Returns", "overlaps", I would remove "e.g.," in brackets since that's how the function is defined.

Comment (nouiz) [reply tensorflow/compiler/xla/window_util.h]: Fixed

**Review by cheshire — DISMISSED** [review]
LGTM - by removing "e.g." I've actually meant keeping the brackets, but just removing "e.g." in front.

**Comment by gbaned** [comment]
@nouiz Could you please address Ubuntu Sanity errors? Thanks!

**Comment by nouiz** [comment]
I tried to fit it, but I don't understand why it do not like it.

**Review by cheshire — APPROVED** [review]

**Pull request merged**